packages/language-server/src/plugins/typescript/features/CompletionProvider.ts

```diff
@@ -910,7 +914,16 @@ export class CompletionsProviderImpl implements CompletionsProvider<CompletionRe
             );
         }
 
-        parts.push(changeSvelteComponentName(ts.displayPartsToString(displayParts)));
+        let text = changeSvelteComponentName(ts.displayPartsToString(displayParts));
+        if (
+            tsDoc.isSvelte5Plus &&
+            text.includes('(alias)') &&
+            text.includes('__SvelteComponent_')
+        ) {
+            // The info contains both the const and type export along with a bunch of gibberish we want to hide
```

Comment:
The `__sveltets_2_IsomorphicComponent` type also appears in hover and the completion documentation when the component is already imported. I am wondering if it makes sense to create a type for it and use it to annotate the `Component__SvelteComponent_` variable. That way the type will be named so it won't show the internal interface and the large generic parameters. But that'll be a bit more complex because we can't use a helper function and need to write JSDoc when in js instead. And also, does having this new type and name actually make things easier to understand?

```ts
const __propsDef = render();
type Input_SvelteComponent_Function = __sveltets_2_IsomorphicComponent<typeof __propsDef['props']>;
const Input__SvelteComponent_: Input_SvelteComponent_Function = __sveltets_2_any({});
```

https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEAzArgOzAFwJYHtXwH0CBnANxAgxA2IICYCticBbHGABwAsswCxWHPCFQYAPACh40+AAUYODsXggAHlVTBlAJXDtgY4hhhZUAcwA08KKgCeAPisBRcqOVqNW+LoEwDRk3MrGwcrAGUIHBoVdREvH31DY1NLaztHeCdVIRhojzidPT8kwNSQjIAhU2AU91jNZQCUiXsACgBraGJiAC54AG8OBSU++UViKxBXGj6XERorYkiZ+AioiZicmgB+Wez2BfgAI2ra3fgqzVr4AF8ASj6iMgoqGnoCAElmNk4eMABhQTCURiMZKZzTDZrQ5ZLYbS41czEewAbgkElMVBgiCgYAQT3IlGotAYX1Y7G4vEBLCEqHmoOGdU8hV8-mSQTSDngAF5ORD5kyCt4imzSsF0jy+atloKGsLWSUUuKubyQs59rllLz+jcrAjrrymuZ7AMpDI6QB3eCtRTYPC9eBYGkHVoAcmeRNddwAdNTafNAagAshMOwAPIcO1BhnjewPR3O3Juj1UL3esKEqh+4HiMEbOZucIyk0AMgG8AAJBWTlcked9UjbvAy7CDsQ0TJrZiQDBUFAIH00O1UDgLagrENxqNGTFmfLEkbUnTyDATdsBlWpgLzgXDlWlutztDiDd4NPxs2NxWt24d5CrPuZUeZfc9nCOzIAF4EKs1xFmYh61OJE0RudEBCDDBCAISclAIUBEElfp4Fgh0dUmSE+nQ+ADxWbC-1qPpXVdSYNTwm4bjRDBbA4BAABUQCMbM6VEAAxNBMFwfBeQJF5iXeMkfkpAEgRY8RqNonBEKIVD4JARAAG1XVQ10AF1UXRCQIKMeAAEE+gYpjRPmdj0CjSVUGQCAIGsZQQg-eAAHpHPgAA9bZwPtKCKklXiiTeBgmHJX5eH4YzRFaGTGTkxA7jNJyXPcoA

Reply:
Good find, I removed it from hover and completion when already imported